client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/ClientInputFile.java

```diff
@@ -44,6 +44,11 @@
    * Charset to be used to read file content. If null it means the charset is unknown and analysis will likely use JVM default encoding to read the file.
    */
   @CheckForNull
-  Charset charset();
+  Charset getCharset();
+
+  /**
+   * Allow clients to pass their own object to ease mapping of issues.
+   */
+  <G> G getClientObject();
```

Comment:
That's what I did at first but code was full of ClientInputFile<?>. Since in fact we don't care about this type I decided to got that way to save the cast for client.